doc/source/user/library/examples.rst

```diff
@@ -966,3 +966,318 @@ modules while in ``u.modules_stats`` the two dimension are switched.
+		The polar decoded bits
+	    """
+	    assert received.shape[1] > 0
+	    return polar_decode_multiple(k, n, frozen_bits, received)
+
+:numref:`codec_hpp` implements the C++ to AFF3CT interface, and it implements the basic
+logic of Polar frozen bits generator, encoder and decoder.
+
+.. code-block:: c++
+	:caption: C++-AFF3CT interface
+	:name: codec_hpp
+	:emphasize-lines: 
+	:linenos:
+
+	#include <iostream>
+	#include <string>
+	#include <vector>
+
+	#include <aff3ct.hpp>
+
+	using namespace aff3ct;
+
+	/**
+	 * @brief Generate frozen bits for a specific SNR
+	 *
+	 * @param k The number of information bits
+	 * @param n The codeword length
+	 * @param snr_max Estimated SNR (in dB)
+	 * @return auto std::vector<bool> of frozen bits
+	 */
+	auto generate_frozen_bits(const int k, const int n, const float snr_max) {
+	  // calculate constants
+	  auto r = static_cast<float>(k * 1.0 / n);
+	  const auto esn0 = tools::ebn0_to_esn0(snr_max, r);
+	  const auto ebn0 = tools::esn0_to_ebn0(esn0);
+	  const auto sigma = tools::esn0_to_sigma(esn0);
+
+	  // set noise
+	  tools::Frozenbits_generator_GA_Arikan frozen_bits_generator(k, n);
+	  auto noise = std::unique_ptr<tools::Sigma<>>(new tools::Sigma<>());
+	  noise->set_values(sigma, ebn0, esn0);
+
+	  // generate frozen bits
+	  std::vector<bool> frozen_bits(n);
+	  frozen_bits_generator.set_noise(*noise);
+	  frozen_bits_generator.generate(frozen_bits);
+	  return frozen_bits;
+	}
+
+	/**
+	 * @brief Polar encoder for single frame
+	 *
+	 * @param k The number of information bits
+	 * @param n The codeword length
+	 * @param frozen_bits std::vector<bool>, frozen bits (length k)
+	 * @param info_bits std::vector<int>, information bits (length k)
+	 * @return auto Encoded codewords, std::vector<int> of length n
+	 */
+	auto polar_encode(const int k, const int n,
+			  const std::vector<bool> &frozen_bits,
+			  const std::vector<int> &info_bits) {
+	  // populate vector
+	  std::vector<int> encoded_bits(n);
+
+	  // encode
+	  module::Encoder_polar<int> polar_encoder(k, n, frozen_bits);
+	  polar_encoder.encode(info_bits, encoded_bits);
+	  return encoded_bits;
+	}
+
+	/**
+	 * @brief Polar encoder for multiple frames
+	 *
+	 * @param k The number of information bits
+	 * @param n The codeword length
+	 * @param frozen_bits std::vector<bool>, frozen bits (length k)
+	 * @param info_bits std::vector<int>, information bits (length k)
+	 * @return auto Encoded codewords, std::vector<int> of length n
+	 */
+	auto polar_encode_multiple(const int k, const int n,
+				   const std::vector<bool> &frozen_bits,
+				   const std::vector<std::vector<int>> &info_bits) {
+	  // populate vector
+	  std::vector<std::vector<int>> encoded_bits;
+
+	  // encode
+	  module::Encoder_polar<int> polar_encoder(k, n, frozen_bits);
+	  for (auto frame : info_bits) {
+	    std::vector<int> encoded(n);
+	    polar_encoder.encode(frame, encoded);
+	    encoded_bits.push_back(encoded);
+	  }
+	  return encoded_bits;
+	}
```

Comment:
I think we can remove this code.

Reply:
No.

Reply:
I think there is too many blocks of code and not so many explanations. If the code is given with no comments, you can simply add a link to the file in the `my_project_with_aff3ct` repo. The other solution is to keep this code here but to split it in smaller blocks and to explain what you are doing in the text.

Reply:
Done. Please refer to the newest commit.